packages/happy-dom/src/fetch/FetchHandler.ts

```diff
@@ -17,14 +20,30 @@ export default class FetchHandler {
 	 * @param [init] Init.
 	 * @returns Response.
 	 */
-	public static fetch(document: IDocument, url: string, init?: IRequestInit): Promise<IResponse> {
+	public static fetch(
+		document: IDocument,
+		url: RequestInfo,
+		init?: IRequestInit
+	): Promise<IResponse> {
 		// We want to only load NodeFetch when it is needed to improve performance and not have direct dependencies to server side packages.
 		const taskManager = document.defaultView.happyDOM.asyncTaskManager;
 
 		return new Promise((resolve, reject) => {
 			const taskID = taskManager.startTask();
 
-			NodeFetch(RelativeURL.getAbsoluteURL(document.defaultView.location, url), init)
+			let request;
+			if (typeof url === 'string') {
+				request = new Request(RelativeURL.getAbsoluteURL(document.defaultView.location, url));
+			} else if (url instanceof URL) {
+				// URLs are always absolute, no need for getAbsoluteURL.
+				request = new Request(url);
```

Comment:
Are you sure this will work if we do `await fetch('my_file.json')`? :sweat_smile:  Because I don't think "node-fetch" will be able to handle it, as it doesn't know where it is. Another thing is that some people might simulate to be on a certain environment by setting location.href to something completely else from the local server (e.g. google.com).

Reply:
The behavior of `await fetch('my_file.json')` should not have changed, the relative URL still gets resolved to an absolute URL as before.

The same logic applies to URLs in Request objects (`await fetch(new Request('my_file.json'))`)

Only for `URL` objects (`await fetch(new URL('http://example.com/my_file.json'))`) it is not necessary to resolve the url, because URL objects are always absolute (see https://developer.mozilla.org/en-US/docs/Web/API/URL/URL#parameters)